src/Components/Activities/ActivitiesForm.js

```diff
@@ -1,31 +1,165 @@
-import React, { useState } from 'react';
+import React, { useState, useRef } from 'react';
 import '../FormStyles.css';
+import { CKEditor } from '@ckeditor/ckeditor5-react';
+import ClassicEditor from '@ckeditor/ckeditor5-build-classic';
+import { FormikProvider, Form, Field, ErrorMessage, useFormik } from "formik";
 
-const ActivitiesForm = () => {
-    const [initialValues, setInitialValues] = useState({
-        name: '',
-        description: ''
-    });
-
-    const handleChange = (e) => {
-        if(e.target.name === 'name'){
-            setInitialValues({...initialValues, name: e.target.value})
-        } if(e.target.name === 'description'){
-            setInitialValues({...initialValues, description: e.target.value})
+const ActivitiesForm = ({
+    activityToEdit
+}) => {
+    const isCreate = activityToEdit===undefined?true:false;
+
+    const fileInput = useRef();
+    const descriptionInput = useRef();
+    const [imageFile, setImageFile ] = useState("");
+    const initialValues = 
+        isCreate?
+            {
+                name: '',
+                image:'',
+                description: ''
+            }
+        :
+        activityToEdit;
+
+    const handleErrors = ( values ) => {
+        let errors = {}
+        if(!values.name){
+            errors.name = "Por favor, ingrese un nombre."
         }
+        if(!values.image){
+            errors.image = "Por favor, ingrese una imagen."
+        }
+
+        return errors;
     }
 
-    const handleSubmit = (e) => {
+    function handleFileChange(e){
         e.preventDefault();
-        console.log(initialValues);
+        if(e.target.files[0]!==undefined){
+            const reader = new FileReader();
+            reader.readAsDataURL(e.target.files[0]);
+            reader.onload = () => {
+                formik.setFieldValue("image",reader.result, true);
+                formik.validateField("image");
+            };
+            reader.onerror = (error)=>{
+                fileInput.current.value = "";
+                formik.setFieldValue("image","",true);
+                formik.validateField("image")
+                alert("No se pudo cargar el archivo de imagen. Error:" + error + ".");
+            }
+        }else{
+            formik.setFieldValue("image","", true);
+            formik.validateField("image");
+        }
+    }
+
+    function handleDescriptionChange(__,editor){
+        const data = editor.getData();
+    }
+
+    function handleDescriptionBlur(e){
+        e.target = {};
```

Comment:
No sé si reescribir los campos del objeto event es el mejor enfoque, al menos nunca ví que se hiciera esto. ¿Hay alguna forma alternativa de lograr los mismo? En general, el objeto event de un event handler se usa como read only.

Reply:
I fixed this by deletting the function altogether. It wasn't necessary because it isn't necessary to validate that field.